src/analysis/region_analysis/region_analysis.py

```diff
@@ -0,0 +1,400 @@
+from pathlib import Path
+import xarray as xr
+import pandas as pd
+from datetime import datetime
+from typing import Tuple, Dict, List, Union, Optional
+import warnings
+import numpy as np
+from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score
+import itertools
+
+from .region_geo_plotter import RegionGeoPlotter
+
+
+class RegionAnalysis:
+    """Create summary statistics for all Regions (defined as xr.Dataset objects)
+    comparing the model predictions against the true values for both train and
+    test datasets.
+
+    This class produces two different formats for the same data. Firstly, it
+    writes a csv file for each model-region object.
+    Secondly, it saves all of the data to a class attribute (`self.df`) in long
+    format. This allows the class to then produce summary statistics.
+
+    Attributes:
+    -----------
+    :region_ds: xr.DataArray
+    :region_lookup: Dict
+    :self.pred_variable: str
+    :self.true_variable: str
+    :self.models_dir: Path
+    :self.features_dir: Path
+    :self.models: List[str]
+
+    TODO:
+    # train or test `true` data ?
+    self.mode = 'test' if test_mode else 'train'
+
+    Note:
+    - because we only save data from the `test` data (vs. the `train` data)
+     we can currently only use data from:
+     `(data/features/{experiment}/test).glob('**/*.nc')` to do our analysis
+    - this doesn't have to be the case if we use the trained model and push the
+    data in `X.nc` through
+    """
+    df: pd.DataFrame
+
+    def __init__(self,
+                 data_dir: Path = Path('data'),
+                 experiment: str = 'one_month_forecast',
+                 admin_boundaries: bool = True):
+        self.pred_variable: Optional[str] = None
+        self.true_variable: Optional[str] = None
+
+        self.data_dir: Path = data_dir
+        self.models_dir: Path = data_dir / 'models' / experiment
```

Comment:
The structure of this directory is
`data_dir/models/experiment/<MODEL_NAME>/preds`

It probably makes sense to include a way to decide which model should be evaluated

Reply:
```python
        if models is not None:
            self.models = self.models[np.isin(self.models, models)]
````

Reply:
where `models` is provided as an initialisation argument

Reply:
Ah I see - thanks :)